### DISCUSSION TITLE
Can Hexo handle 10M pages?

### DISCUSSION THREAD

**Discussion by drez25** [hexojs/hexo#4990] [Question / Answer]
Hello,
I have a website with 10M pages currently running on WordPress and it is slow. 
Can hexo handle 10M files efficiently and quickly?

**Comment by lorezyra** [comment]
HEXO architecture requires that the entire source folder be loaded into memory... Any machine with enough RAM should be able to handle that. Also, you probably need to configure the filesystem to allow more than 10K files to be opened at once. Linux machines can probably do that. But for that many pages with unknown number of assets (images, etc) would likely require a large server to generate the website.

Once generated, the site will load blazingly fast.

**Comment by stevenjoezhang** [comment]
See also https://github.com/hexojs/hexo/issues/2579#issuecomment-1328065978
